src/sinks/signalfx/metrics.rs

```diff
@@ -0,0 +1,368 @@
+        ///*
+        /// Generated by analytics server
+        Job = 300000,
+        ///*
+        /// @deprecated
+        /// Event originated within collectd (deprecated in favor of AGENT)
+        Collectd = 400000,
+        ///*
+        /// Service discovery event
+        ServiceDiscovery = 500000,
+        ///*
+        /// Created by exception appenders to denote exceptional events
+        Exception = 700000,
+        ///*
+        /// Event originated from an agent
+        Agent = 2000000,
+    }
+    //}
+}
+
+pub fn default_realm() -> String {
+    String::from("us0")
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct SignalFxConfig {
+    // TODO: namespace a du sens dans le contexte SignalFx ? il préfixe les métriques
+    pub namespace: String,
+    #[serde(default = "default_realm")]
+    pub realm: String,
+    pub token: String,
+    #[serde(default)]
+    pub batch: BatchEventsConfig,
+    #[serde(default)]
+    pub request: TowerRequestConfig,
+}
+
+struct SignalFxSink {
+    config: SignalFxConfig,
+    last_sent_timestamp: AtomicI64,
+    uri: Uri,
+}
+
+lazy_static! {
+    static ref REQUEST_DEFAULTS: TowerRequestConfig = TowerRequestConfig {
+        retry_attempts: Some(5),
+        ..Default::default()
+    };
+}
+
+inventory::submit! {
+    SinkDescription::new::<SignalFxConfig>("signalfx_metrics")
+}
+
+#[typetag::serde(name = "signalfx_metrics")]
+impl SinkConfig for SignalFxConfig {
+    fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        let healthcheck = healthcheck(self.clone(), cx.resolver())?;
+
+        let batch = self.batch.unwrap_or(20, 1);
+        let request = self.request.unwrap_with(&REQUEST_DEFAULTS);
+
+        let uri = format!("https://ingest.{}.signalfx.com/v2/datapoint", self.realm)
+            .parse::<Uri>()
+            .context(super::UriParseError)?;
+        let timestamp = Utc::now().timestamp();
+
+        let sink = SignalFxSink {
+            config: self.clone(),
+            uri,
+            last_sent_timestamp: AtomicI64::new(timestamp),
+        };
+
+        let sink = BatchedHttpSink::new(sink, MetricBuffer::new(), request, batch, None, &cx)
+            .sink_map_err(|e| error!("Fatal SignalFx error: {}", e));
+
+        Ok((Box::new(sink), healthcheck))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Metric
+    }
+
+    fn sink_type(&self) -> &'static str {
+        "signalfx_metrics"
+    }
+}
+
+impl HttpSink for SignalFxSink {
+    type Input = Event;
+    type Output = Vec<Metric>;
+
+    fn encode_event(&self, event: Event) -> Option<Self::Input> {
+        Some(event)
+    }
+
+    fn build_request(&self, events: Self::Output) -> http::Request<Vec<u8>> {
+        let now = Utc::now().timestamp();
+        let interval = now - self.last_sent_timestamp.load(Acquire);
+        self.last_sent_timestamp.store(now, Release);
```

Comment:
This looks like a race condition. Should it use a loop around `compare_and_swap` or `compare_exchange` instead?
```rust
let mut now = Utc::now().timestamp();
while self.last_sent_timestamp.compare_and_swap(now, AcqRel) != now {
    now = Utc::now().timestamp();
}
```